src/main/scala/nl.knaw.dans.easy.deposit/docs/StateInfo.scala

```diff
@@ -23,14 +23,20 @@ import scala.collection.Seq
 import scala.util.{ Failure, Success, Try }
 
 case class StateInfo(state: State, stateDescription: String) {
-  def isDeletable: Try[Unit] = {
+  def canDelete: Try[Unit] = {
     if (StateInfo.deletableStates.contains(state)) Success(())
     else Failure(new IllegalStateException(s"Deposit has state $state, can only delete deposits with one of the states: ${ StateInfo.deletableStates.mkString(", ") }"))
   }
+
+  def canUpdate: Try[Unit] = {
```

Comment:
1. `canUpdate` and `canDelete` look awfully similar. If you have a third `canXXX`, I would suggest refactoring them a bit.
2. These error messages are very descriptive. Is there any chance they end up in {{easy-deposit-ui}}. @jo-pol could you take these into account when you look at EASY-2016?

Reply:
In theory these errors could be caused by impatient users. For example delete a deposit on one browser tab while a submission on the same deposit is in progress on another. More likely by clients that don't look at the state. So the client should know these details. 

Issue 2016 is more about details like absolute file paths.

Reply:
@ 1) I suppose you mean some private `can(action: String, states: Seq[State])`